src/ndarray/ndarray.cc

```diff
@@ -223,6 +223,80 @@ void ScalarOp(const NDArray &lhs,
   }
 }
 
+void AssignSliceFrom(const NDArray &from, int slice_dim, index_t start, index_t end,
```

Comment:
Will leave it for now and we can replace it after we have a partial assign operator.